src/charm.py

```diff
@@ -112,6 +114,22 @@ def _send_info(self, interfaces, secret_key):
                 }
             )
 
+    def _get_minio_args(self):
+        model_mode = self.model.config["mode"]
+        if model_mode == "server":
+            return ["server", "/data"]
+        elif model_mode == "gateway":
+            storage = self.model.config["gateway_storage_service"]
```

Comment:
should we validate this here?  Do we know enough to validate it?  At a minimum, do we know this will always need to be non-empty (eg: will it fail if someone does `server=gateway` but forgets to set `gateway_storage_service`?  If so, maybe we catch that here with a helpful status message

Reply:
Added early check.